### PR TITLE
PLT-2962 Added channel switcher modal

### PR DESCRIPTION
**Pull request by DavidLu1997**
CTRL/CMD+K now gets you to a channel switcher dialog, allowing you to select a non-DM channel using autocomplete!

The `Go` button now says `Switch`

### CODE REVIEW

File: webapp/components/channel_switch_modal.jsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import SuggestionList from './suggestion/suggestion_list.jsx';
+import SuggestionBox from './suggestion/suggestion_box.jsx';
+import SwitchChannelProvider from './suggestion/Switch_channel_provider.jsx';
```
Comment (enahum) [inline webapp/components/channel_switch_modal.jsx]: this import should be in lowecase

File: webapp/components/channel_switch_modal.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                                defaultMessage='Switch Channels'
+                            />
+                        </span>
+                    </Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>
+                    <FormattedMessage
+                        id='channel_switch_modal.help'
+                        defaultMessage='`tab` or `↑↓` to browse, `↵` to select, `esc` to dismiss'
```
Comment (enahum) [inline webapp/components/channel_switch_modal.jsx]: This text should be the same as the one in en.json `channel_switch_modal.help`

**Comment by DavidLu1997** [comment]
Fixed :)

**Comment by jasonblais** [comment]
1. Unable to use up/down arrows to browse
2. When there are two channel names with the same display name (but different URL), only one is displayed

image

**Comment by jasonblais** [comment]
+1 for design

**Comment by enahum** [comment]
@DavidLu1997 did you fixed the problems reported by @jasonblais ?

**Comment by DavidLu1997** [comment]
Yep, fixed everything

**Comment by enahum** [comment]
+1

**Pull request merged**